File: Easy/0026_Remove_Duplicates_from_Sorted_Array/c++/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = nums.size();
        
        if (nums.size() == 0) {
            return 0;
        }
        
        for (int i = 0; i < nums.size()-1;) {
            if (nums[i] == nums[i+1]) {
                nums.erase(nums.begin()+i);
                length--;
            } else {
                i++;
            }
        }
        
        return length;
    }
};

File: Easy/0026_Remove_Duplicates_from_Sorted_Array/python3/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        currentNumber = 0
        pointer = 1
        duplicatesFound = 0
        
        while pointer < len(nums):
            if nums[pointer] == nums[currentNumber]:
                nums[pointer] = -999
                duplicatesFound = duplicatesFound + 1
            else:
                if duplicatesFound > 0:
                    nums[currentNumber+1] = nums[pointer]
                    nums[pointer] = -999
                    duplicatesFound = 0
                    currentNumber = currentNumber + 1
                elif duplicatesFound == 0:
                    nums[currentNumber+1] = nums[pointer]
                    currentNumber = currentNumber + 1
            pointer = pointer + 1
        return currentNumber+1